src/core/cosmopolite.scala

```diff
@@ -11,18 +11,20 @@ object Language:
    @targetName("make")
    def apply[L <: String: ValueOf]: Language[L] = new Language(summon[ValueOf[L]].value)
    
-   inline def parse[L <: String](str: String): Option[Language[L]] = ${doParse[L]('str)}
+   inline def parse[L <: String](str: String): Option[Language[L]] =
+      Option.when(reifyToList[L].contains(str))(Language(str))
 
-   private def doParse[L <: String: Type](str: Expr[String])(using quotes: Quotes): Expr[Option[Language[L]]] =
+   inline def reifyToList[L <: String]: List[String] = ${reifyToListMacro[L]}
+
+   private def reifyToListMacro[L <: String: Type](using quotes: Quotes): Expr[List[String]] =
       import quotes.reflect._
 
       def langs(t: TypeRepr): List[String] = t.dealias match
          case OrType(left, right) => langs(left) ++ langs(right)
+         case AndType(left, right) => langs(left) ++ langs(right)
```

Comment:
I'm not sure I want to give meaning to an expression like, `En & Fr | De & Es` which would be equivalent to `En | Fr | De | Es` according to the `reifyToList` macro, while the types would not conform. It seems easier to be less permissive in the representation.